File: jelly-runtime/src/store/dapp/access/token_balance.ts
import { Chain } from './chain';

export interface TokenBalance {
    chain: Chain; // chain
    address?: string; // Main currency or designated token
    balance: number; // Minimum ownership
}

export interface VerifiedTokenBalance {
    chain: Chain; // chain
    address?: string; // Main currency or designated token
    balance: number; // Minimum ownership

    // =============== Verification information ===============
    identity: string; // Identity address
    message: string;
    signature: string;
}

File: jelly-model/src/store/dapp/access/token_balance.rs
use serde::{Deserialize, Serialize};

/// Token balance
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct TokenBalance {
    /// chain
    pub chain: super::chain::Chain,
    /// Main currency or designated token
    #[serde(skip_serializing_if = "Option::is_none")]
    pub address: Option<String>, // Specify tokens, or main currency
    /// Minimum quota
    pub balance: u64, // Minimum ownership
}

/// Tokens balance verification
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct VerifiedTokenBalance {
    /// chain
    pub chain: super::chain::Chain,
    /// Main currency or designated token
    #[serde(skip_serializing_if = "Option::is_none")]
    pub address: Option<String>, // Specify tokens, or main currency
    /// Minimum quota
    pub balance: u64, // Minimum ownership

    // =============== Verification information ===============
    /// Identity address
    pub identity: String,
    /// Signature information
    pub message: String,
    /// signature
    pub signature: String,
}
